.github/ALLOWLIST

```diff
@@ -31,4 +31,5 @@ pkg/experimentalmetricmetadata
 processor/deltatorateprocessor
 processor/metricsgenerationprocessor
 receiver/dotnetdiagnosticsreceiver
+receiver/snowflakereceiver
```

Comment:
Let's add it in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS instead of the ALLOWLIST